blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/LocatorFetchRunnable.java

```diff
@@ -102,10 +104,25 @@ public void run() {
 
         Set<Locator> locators = getLocators(executionContext);
 
+        boolean isSlotBeingRerolled = scheduleCtx.isReroll(parentSlotKey);
         for (Locator locator : locators) {
-            rollCount = processLocator(rollCount, executionContext, rollupBatchWriter, locator);
+
+            //if slot is being re-rolled, only process locator if locator's lastUpdateTimestamp is after last rollup time
+            if (IS_REROLL_ONLY_DELAYED_METRICS &&
+                    isSlotBeingRerolled) {
+
+                UpdateStamp updateStamp = scheduleCtx.getShardStateManager().getUpdateStamp(parentSlotKey);
+                if (locator.getLastUpdatedTimestamp() > updateStamp.getLastRollupTimestamp()) {
+                    rollCount = processLocator(rollCount, executionContext, rollupBatchWriter, locator, true);
+                }
+
+            } else {
+                rollCount = processLocator(rollCount, executionContext, rollupBatchWriter, locator, isSlotBeingRerolled);
+                Instrumentation.markLocatorRolled();
```

Comment:
There is a call to Instrumentation.markLocatorReRolled() inside processLocator(). Is this double counting it?